src/mpiwrap/message_passing.F

```diff
@@ -1140,11 +1143,23 @@ SUBROUTINE mp_world_init(mp_comm)
             IF (ierr /= 0) CALL mp_stop(ierr, "mpi_init @ mp_world_init")
 !$       ELSE
 !$OMP MASTER
+#if defined(__DLAF)
+            ! DLA-Future requires that the MPI library supports
+            ! THREAD_MULTIPLE mode
+!$          CALL mpi_init_thread(MPI_THREAD_MULTIPLE, provided_tsl, ierr)
```

Comment:
Oh, I didn't see this before. That a pretty major constraint, isn't it?

While I agree that `MPI_THREAD_MULTIPLE` should be the norm these days, the last time I checked the support for it was still rather poor.

Reply:
This is a requirement for DLA-Future. I updated the error message below to reflect this.